### DISCUSSION TITLE
SH-ESP32 dual use

### DISCUSSION THREAD

**Discussion by jonasbager** [hatlabs/discussions#103] [General]
Hi. I am in the middle of setting up my SH-ESP32 board to be used for various sensors around the boat. But I also saw the example tutorial for setting it up as a NMEA2000 wifi gateway. My question is: Can it be used as both at the same time?

Secondary question: With the new Halmet being launched, is this board still ideal for measuring sensors (for example from the engine) or should I also get the Halmet? 
Thanks for the help

**Comment by mairas** [comment]
> My question is: Can it be used as both at the same time?

I guess it's possible but it's not going to be straightforward or practical. Debugging becomes complex because the standard serial port is reserved for N2K traffic, and there might be some complications in transmitting NMEA 2000 traffic.

> Secondary question: With the new Halmet being launched, is this board still ideal for measuring sensors (for example from the engine) or should I also get the Halmet?

SH-ESP32 is more generic than HALMET. If you need to interface e.g. some additional sensor boards, SH-ESP32 has more available GPIOs and the proto board area to play with. But HALMET makes things much easier if you need to measure analog inputs or digital inputs with wide voltage ranges. It really depends on the extra effort you're willing to make, but for interfacing an engine or tank sensors specifically, HALMET is the thing.

**Comment by bradharley** [comment]
I just asked a similar question on the SH-wg github.   Has anyone tried this and/or made progress?   Debugging would be harder, but you could always dump the debug to a tcp stream rather than the USB port.   That would work except for bootstrap and I normally do that on my esp projects.  I would imagine that at a minimum dumping 0183 stream over TCP/UDP would be doable if not the full implementation...
